geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -1215,25 +1216,26 @@ public List peek(int batchSize, int timeToWait) throws InterruptedException, Cac
     final boolean isDebugEnabled = logger.isDebugEnabled();
 
     PartitionedRegion prQ = getRandomShadowPR();
-    List<GatewaySenderEventImpl> batch = new ArrayList<>();
     if (prQ == null || prQ.getLocalMaxMemory() == 0) {
       try {
         Thread.sleep(50);
       } catch (InterruptedException e) {
         Thread.currentThread().interrupt();
       }
       blockProcessorThreadIfRequired();
-      return batch;
+      return Collections.EMPTY_LIST;
     }
+    List<GatewaySenderEventImpl> batch =
+        new ArrayList<>(batchSize == BATCH_BASED_ON_TIME_ONLY ? DEFAULT_BATCH_SIZE : batchSize);
```

Comment:
This will create array of capacity 100 (DEFAULT_BATCH_SIZE). Earlier the array growth was taken care by java. If there are not many events in the queue, it may be overhead.

Reply:
This was an attempt to make serial and parallel implementations more consistent.  Javas default size is quite small and I would guess that most of the time we are having to grow to the "batchSize" that is configured anyways?